kernel/kernel/tty/tty.cpp

```diff
@@ -297,6 +297,29 @@ size_t ttydevfs_write(size_t offset, size_t len, void *ubuffer, struct file *f)
     return len;
 }
 
+ssize_t ttydevfs_write_iter(file *filp, size_t offset, iovec_iter *iter, unsigned int flags)
+{
+    struct tty *tty = (struct tty *) filp->f_ino->i_helper;
+
+    char *buffer = (char *) malloc(iter->bytes);
+    if (!buffer)
+    {
+        return (size_t) -ENOMEM;
+    }
+
+    ssize_t copied = copy_from_iter(iter, buffer, iter->bytes);
+    if (copied < 0)
+    {
+        free(buffer);
+        return -EFAULT;
+    }
+
+    tty_write(buffer, copied, tty);
```

Comment:
I think it is possible to make tty_write_iter function, so we can avoid allocating additional buffer

Reply:
Yes, this is required, allocating a buffer is 1) too slow and 2) a possible DoS point given it takes a 64-bit user-controlled integer